Sources_v3/Utils/Internet Connection/InternetConnection.swift

```diff
@@ -0,0 +1,250 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import Network
+
+/// A delegate to receive Internet connection events.
+protocol InternetConnectionDelegate: class {
+    /// Calls when the Internet connection status did change.
+    /// - Parameter status: an Internet connection status.
+    func internetConnectionStatusDidChange(status: InternetConnection.Status)
+    
+    /// Calls when the Internet connection monitor got an error.
+    /// For example `Reachability` didn't initiated or started notifying properly.
+    /// - Parameter error: an error.
+    func internetConnectionMonitorFailed(with error: Error)
+}
```

Comment:
You're gonna hate me, but I don't think `delegate` is the best pattern for this :) I think the model `Reachability` uses (posting updates using `NotificationCenter`) is more suitable for this.

Reply:
Fixed